tgg/seqr_scripts/sex_check.py

```diff
@@ -0,0 +1,199 @@
+    sex_ht = hl.impute_sex(mt.GT, aaf_threshold=aaf_threshold, male_threshold=male_fstat_threshold, female_threshold=female_fstat_threshold)
+    mt = mt.annotate_cols(**sex_ht[mt.col_key])
+    sex_ht = mt.cols()
+    
+    # plot histogram of fstat values
+    df = sex_ht.to_pandas()
+    plt.clf()
+    plt.hist(df["f_stat"])
+    plt.xlabel("Fstat")
+    plt.ylabel("Frequency")
+    plt.axvline(male_fstat_threshold, color='blue', linestyle='dashed', linewidth=1)
+    plt.axvline(female_fstat_threshold, color='red', linestyle='dashed', linewidth=1)
+
+    outplot = f"{outdir}/fstat_{callset_name}.png"
+    with hl.hadoop_open(outplot, 'wb') as out:
+        plt.savefig(out)   
+        
+    return sex_ht  
+
+    
+def call_sex(callset: str, 
+             use_y_cov: bool=False,
+             y_cov_threshold: float=0.1,
+             normalization_contig: str = "chr20",
+             male_fstat_threshold: float=0.75, 
+             female_fstat_threshold: float=0.5, 
+             aaf_threshold: float=0.05,
+             call_rate_threshold: float=0.25) -> hl.Table:
+    
+    """
+    Calls sex for the samples in a given callset and exports results file to the callset directory
+    :param callset: String of full matrix table path for the callset
+    :param use_y_cov: Set to True to calculate and use chrY coverage for sex inference
+    :param y_cov_threshold: Coverage on chrY above which supports male call
+    :param male_fstat_threshold: Fstat threshold above which a sample will be called male
+    :param female_fstat_threshold: Fstat threshold below which a sample will be called female
+    :param aaf_threshold: Minimum alternate allele frequency required 
+    :param call_rate_threshold: Minimum required call rate
+    :return: Table with sex annotations
+    """
+
+    # read in matrix table and define output directory
+    # will need to generalize more for gnomad methods
+    outdir = dirname(callset)
+    mt_name = callset.split("/")[-1].strip("\.mt")
+    logging.info(f"Reading matrix table for callset: {callset}")
+    logging.info(f"Using chromosome Y coverage? {use_y_cov}")
+
+    mt = hl.read_matrix_table(callset)
+    
+    # filter to SNVs and biallelics
+    # filter to biallelics might be a function
+    mt = mt.filter_rows((hl.len(mt.alleles) == 2) & hl.is_snp(mt.alleles[0], mt.alleles[1]))
+    
+    # filter to pass variants only (empty set)
+    # gnomad methods - will need to make this an optional argument
+    mt = mt.filter_rows(mt.filters.length() == 0, keep=True)
+    
+    #infer build:
+    build = get_reference_genome(mt.locus).name
+    logging.info(f"Build inferred as {build}")
+    
+    logging.info("Inferring sex...")
+    #for production change female and male to XX and XY
+    if use_y_cov:
+        mt = mt.annotate_cols(**{f"{normalization_contig}_mean_dp": get_chr_cov(mt, build, call_rate_threshold, normalization_contig)
+        "chrY_mean_dp": get_chr_cov(mt, build, call_rate_threshold, chrY_name)})
```

Comment:
```suggestion
        mt = mt.annotate_cols(
            **{
                f"{normalization_contig}_mean_dp": get_chr_cov(mt, build, call_rate_threshold, normalization_contig),
                "chrY_mean_dp": get_chr_cov(mt, build, call_rate_threshold, chrY_name),
            }
        )
```
(making this easier to read)

Reply:
also, note that this annotation won't work as expected with the current way `get_chr_cov` is set up!